src/tooltip.tsx

```diff
@@ -135,10 +142,64 @@ const Tooltip: React.FC<Props> = ({
         height: 0,
     });
 
-    const getPosition = (position: Position = defaultPositionDesktop) =>
-        isTabletOrSmaller && (position === 'left' || position === 'right') ? defaultPositionMobile : position;
+    const [containerPosition, setContainerPosition] = React.useState({});
+
+    const getPosition = (position: Position | undefined) =>
+        isTabletOrSmaller ? position || defaultPositionMobile : position || defaultPositionDesktop;
+
+    const noRightSpace = (position: Position) => {
+        if (!tooltipRef.current) return position;
+        const tooltipBoundingClientRect = tooltipRef.current.getBoundingClientRect();
+
+        return targetBoundingClientRect.current.right + tooltipBoundingClientRect.width > window.innerWidth;
+    };
+
+    const noLeftSpace = (position: Position) => {
+        if (!tooltipRef.current) return position;
+        const tooltipBoundingClientRect = tooltipRef.current.getBoundingClientRect();
+
+        return targetBoundingClientRect.current.left < tooltipBoundingClientRect.width;
+    };
+
+    const validatePosition = (position: Position) => {
+        if (!tooltipRef.current) return position;
+        const tooltipBoundingClientRect = tooltipRef.current.getBoundingClientRect();
+
+        const validateLeft = (position: Position) => {
+            const noTopSpace =
```

Comment:
this variable has a bad name, what does this mean? I'd call it something like `alternativePosition`

```ts
return hasLeftSpace() ? 'left' : alternativePosition;
```

Reply:
validateLeft and validateRight, are validating if there is space on the screen to the left and right, depending on whether it has, it will return the position that was passed.

In validate Left checks if there is no space on the left and also not on top, it would end up going to bottom.

Reply:
I mean the `noTopSpace` variable is a bad name